htdocs/compta/tva/class/tva.class.php

```diff
@@ -163,7 +163,7 @@ public function create($user)
 		$now = dol_now();
 
 		// Clean parameters
-		$this->amount = trim($this->amount);
+		$this->amount = (float) $this->amount;
```

Comment:
I've been "PHPDoc typing" in several cases that monetary values can be either string or float because price2num will ensure that they have the desired count of digits after the decimal dot.

Reply:
Yes, that's what I figured out with the phan warnings.
So I've updated to:
```
	/**
	 * @var float|string VAT amount
	 */
	public $amount;
```
and
``` 
// Clean parameters
$this->amount = (float) price2num($this->amount);
```

Reply:
Actually, you can keep 
```
$this->amount = price2num($this->amount);
```

And whenever $this->amount is used in a computation, cast it with (float).

It would be better to have a class represent monetary and quantity values IMHO, but that's a task in itself.  That class would be able to handle all the operators +, -, /, ... .